### PR TITLE
Move the hash_state module to be private

### PR DESCRIPTION
**Pull request by zachmullen**
Adding this to the 3.0 release milestone because it's a breaking change.

This module should not be part of the API surface. I would go ahead and just replace it with [hashstate](https://pypi.org/project/hashstate/) but I haven't had the time to make that work with python 2.7 (that will require some C preprocessor stuff to work with both major versions of CPython's C API). If we decide in the future that we are dropping support for py2.7, it's a very simple switch.

### CODE REVIEW

**Comment by zachmullen** [comment]
Anyone know why a couple of the web tests are failing with the following message?

`Error: Cannot find module 'event-source/global'`

**Comment by manthey** [comment]
They've been failing for about a week, but I don't know why.

**Comment by zachmullen** [comment]
The event-source thing was a local problem, not the real issue.

And of course, these are passing for me locally. I'll try to blow away the circleCI caches.

**Comment by zachmullen** [comment]
The problem is somewhere in the jobs plugin web client bundle, it seems to be attempting to load a script into the page via an XHR through jquery and the jquery is `eval`ing it and barfing. I'm sifting through what exactly that is now...

**Comment by zachmullen** [comment]
@girder/developers  this is ready for review

File: docs/migration-guide.rst [hunk 1]
```diff
@@ -313,8 +313,9 @@ Server changes
     * All of the methods in ``girder.utility.plugin_utilities`` no longer accept a ``curConfig``
       argument since the configuration is no longer read.
 
-* The ``girder.utility.sha512_state`` module has been removed. All of its symbols had been deprecated
-  and replaced by corresponding ones in ``girder.utility.hash_state``.
+* The ``girder.utility.sha512_state`` module has been removed.k
```
[the remark this answers is not in the document]
Comment (zachmullen) [reply docs/migration-guide.rst]: Thanks, fixed

**Pull request merged**